### PR TITLE
feat(api): Expose uptime data as text via API

### PR DESCRIPTION
**Pull request by JamesHillyard**
## Summary
Exposes the uptime data in it's raw format as plaintext in the API for third-party consumption. Implements feature request #757 

_This code is very similar to the code used to fetch the data in `/api/badges.go`_

## Checklist
- [X] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

**Review by TwiN — CHANGES_REQUESTED** [review]
Just FYI, I made a few changes in #759 that breaks your code, namely, the `core` package has been moved to `config/endpoint`.

Sorry about the inconvenience 😅

**Comment by JamesHillyard** [comment]
No worries, thanks for letting me know! Hopefully it's all sorted 😄

File: api/raw.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	if err != nil {
+		if errors.Is(err, common.ErrEndpointNotFound) {
+			return c.Status(404).SendString(err.Error())
+		} else if errors.Is(err, common.ErrInvalidTimeRange) {
+			return c.Status(400).SendString(err.Error())
+		}
+		return c.Status(500).SendString(err.Error())
+	}
+
+	c.Set("Content-Type", "text/html")
```
Comment (TwiN) [inline api/raw.go:37]: Shouldn't this be `text/plain`?

File: api/raw.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	uptime, err := store.Get().GetUptimeByKey(key, from, time.Now())
+	if err != nil {
+		if errors.Is(err, common.ErrEndpointNotFound) {
+			return c.Status(404).SendString(err.Error())
+		} else if errors.Is(err, common.ErrInvalidTimeRange) {
+			return c.Status(400).SendString(err.Error())
+		}
+		return c.Status(500).SendString(err.Error())
+	}
+
```
Comment (TwiN) [inline api/raw.go:36]:
```suggestion
```

**Review by TwiN — CHANGES_REQUESTED** [review]

File: api/raw.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/TwiN/gatus/v5/storage/store"
+	"github.com/TwiN/gatus/v5/storage/store/common"
+	"github.com/gofiber/fiber/v2"
+)
+
+func UptimeRaw(c *fiber.Ctx) error {
+	duration := c.Params("duration")
+	var from time.Time
+	switch duration {
```
Comment (TwiN) [inline api/raw.go:16]: It should be possible to support 30d as well now. Sorry, I had missed the notification.

**Review by TwiN — CHANGES_REQUESTED** [review]

File: api/raw.go (hunk 3 above)
Comment (JamesHillyard) [reply api/raw.go:16]: @TwiN Updated to support 30d uptime 😄

**Comment by aaruni96** [comment]
*bump* I'd be excited if this were merged in

File: README.md [hunk 4]
```diff
@@ -2404,6 +2404,23 @@ Gzip compression will be used if the `Accept-Encoding` HTTP header contains `gzi
...
+Gatus exposes the raw data for one of your monitored endpoints.
+This allows you to track and aggregate data in your own applications for monitored endpoints. For instance if you want to track uptime for a period longer than 7 days.
+
+##### Uptime
+The path to get raw uptime data for an endpoint is:
+```
+/api/v1/endpoints/{key}/uptimes/{duration}
+```
+Where:
+- `{duration}` is `7d`, `24h` or `1h`
```
Comment (TwiN) [inline README.md:2417]:
```suggestion
- `{duration}` is `30d` (alpha), `7d`, `24h` or `1h`
```

**Review by TwiN — APPROVED** [review]

**Pull request merged**